### PR TITLE
Calendar components from:to: date fix

### PR DESCRIPTION
**Pull request by e78l**
With basic single-component (second) test

Issue reported by Malcolm Barclay via swift-corelibs-dev

### CODE REVIEW

**Comment by parkera** [comment]
@swift-ci please test and merge

**Comment by e78l** [comment]
Perhaps ninja needs to be checked on ubuntu 16.04-6? But not sure how https://github.com/apple/swift-corelibs-foundation/pull/818 didn't encounter... (https://ci.swift.org/job/swift-corelibs-foundation-PR-Linux/571/)
CONFLICT (content): Merge conflict in src/ninja.cc
Patch failed at 0001 mark release 120715
The copy of the patch that failed is found in: .git/rebase-apply/patch

When you have resolved this problem, run "git rebase --continue".
If you prefer to skip this patch, run "git rebase --skip" instead.
To check out the original branch and stop rebasing, run "git rebase --abort".

error: Failed to merge in the changes.

Error on repo "/home/buildnode/jenkins/workspace/swift-corelibs-foundation-PR-Linux@3/ninja": Traceback (most recent call last):
  File "/home/buildnode/jenkins/workspace/swift-corelibs-foundation-PR-Linux@3/swift/utils/update-checkout", line 73, in update_single_repository
    shell.run(["git", "rebase", "FETCH_HEAD"], echo=True)
  File "/home/buildnode/jenkins/workspace/swift-corelibs-foundation-PR-Linux@3/swift/utils/swift_build_support/swift_build_support/shell.py", line 215, in run
    raise eout
Exception: ['git', 'rebase', 'FETCH_HEAD']

**Comment by parkera** [comment]
cc @erg @shahmishal

**Comment by erg** [comment]
I made a change to ``update-checkout`` last week to clone ninja from github and check out the release branch. Some of linux test machines have a known buggy ninja 1.5.x, for instance on ubuntu 14.04, because that's all they package.

The git commands we are using are somewhat weak--they should, instead of pulling, fetch and then reset to the remote branch. However, this would possibly overwrite local development.

So the alternative is to clean the workspaces and delete any previous ninja directories, which I guess didn't get done everywhere?

TLDR - probably need to clean this machine's checked-out directories.

**Comment by shahmishal** [comment]
This is failing because update-checkout command was missing `--reset-to-remote`

I just updated the job, it should be fixed now. 

@swift-ci please test and merge

**Comment by ikesyo** [comment]
Could you please re-trigger other recent pull requests those failed for the same reason?

**Comment by UberJason** [comment]
Hello, I'm wondering if this fix is going to be merged into the Swift 3.1 branch before Swift 3.1 is released? As of this morning this fix doesn't appear to be there: https://github.com/apple/swift-corelibs-foundation/blob/swift-3.1-branch/Foundation/NSCalendar.swift

**Comment by alblue** [comment]
It will need to be manually cherry-picked onto the swift-3.1 branch with a new pull request -- it won't happen automatically.
